Allow users to make the scrollbars in the title area larger

JohnnyCrazy opened the pull request:
<!-- Thank you for submitting a Pull Request. Please: 
* Read our Pull Request guidelines:
  https://github.com/Microsoft/vscode/wiki/How-to-Contribute#pull-requests.
* Associate an issue with the Pull Request.
* Ensure that the code is up-to-date with the `master` branch.
* Include a description of the proposed changes and how to test them. 
-->

This PR fixes #32192
It introduces two new settings:
* `workbench.editor.tabScrollbarHeight` - Specify the size of the scrollbar used in the Tab list
* `breadcrumbs.scrollbarHeight` - Specify the size of the scrollbar used in the breadcrumbs list

It updates in real-time, a small demo can be seen here:
![image](https://i.imgur.com/YbNgFVB.gif)

Implementation wise, it's my first code change in VSCode, so I'm quite new to the architecture and I'm not sure if I chose the correct locations. Especially the `constructor` change in `breadcrumbsWidget.ts`. I'm looking forward to your feedback :)

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/vscode?pullRequest=92720) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vscode?pullRequest=92720) before we can accept your contribution.<br/><br/>:x: JohnnyCrazy [sign now](https://cla.opensource.microsoft.com/microsoft/vscode?pullRequest=92720)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vscode?pullRequest=92720) it.</sub>

bpasero commented:
Thanks, this seems to work pretty well. Adding @alexdima for the changes on the scrollbar itself and @jrieken for breadcrumb. 

I wonder if we should have just 1 setting for breadcrumb and tabs, however this may be challenging because after all tabs have more height available compared to breadcrumbs. 

@JohnnyCrazy one early feedback I have is to add an upper limit for the scrollbar height so that people do not end up with the scrollbar taking the entire space. E.g. something like 50% of the height we set for the parent container.

jrieken commented:
Idk - this a good idea? Do we want settings for each scrollbar? What about the peek-errors - users have complaint about that one too. Isn't the better approach to have one scrollbar height setting?

bpasero commented:
@jrieken yeah but as far as I know, we use a consistent size for the scrollbar in all other places, only the editor tabs and breadcrumbs use a smaller height. 

Fair though, we probably need a setting for all other scrollbars too if we wanted to be consistent.

jrieken commented:
> Fair though, we probably need a setting for all other scrollbars too if we wanted to be consistent.

If breadcrumbs and tabs are really the only two special height cases that we should maybe have a setting for that, or a setting for the "special height" scrollbar. I wouldn't allow a number but something symbolic, like "small" and "default".

bpasero commented:
Yeah I like that, we could have then just have 1 setting for both scrollbars. `default` would be as today and `large` could then actually just pick the size we have in all other places of the workbench.

JohnnyCrazy commented:
I think a setting for each scrollbar would be too much hassle to maintain/configure for both, the user and the developer.

> If breadcrumbs and tabs are really the only two special height cases that we should maybe have a setting for that, or a setting for the "special height" scrollbar. I wouldn't allow a number but something symbolic, like "small" and "default".

I like the idea of having a symbolic setting and I could imagine most users who complained about the small tab/breadcrumbs scrollbar don't want to specify a specific pixel number; they just want it to be better grabbable. This would also remove the need of a max height boundary and different parent sizes are no problem.

The naming of such a setting is not easy though, `workbench.specialScrollbarHeight` could be hard to find?

> Yeah I like that, we could have then just have 1 setting for both scrollbars. default would be as today and large could then actually just pick the size we have in all other places of the workbench.

Isn't the size of all other scrollbars a little bit too large for the breadcrumbs list? It would be 14px.

![image](https://user-images.githubusercontent.com/3109892/76743692-513be080-6773-11ea-9966-ce9f2465229b.png)


If we find some consent, would be happy to refactor the PR :)

bpasero commented:
Yeah, I would pick a sensible default for the larger scrollbar that works. Maybe `workbench.editor.titleScrollbarSizing: default | large` ?

JohnnyCrazy commented:
If we limit the setting to the title bar, sounds good

@jrieken mentioned peek-errors, should they also be included in this change? then we would need a different name

bpasero commented:
I would not open the door to all the other places just yet.

JohnnyCrazy commented:
I updated the PR and introduced `workbench.editor.titleScrollbarSizing: default | large`

Large defaults:


![image](https://user-images.githubusercontent.com/3109892/76868556-672acd80-6867-11ea-8220-c6ca77567b71.png)

bpasero commented:
Thanks @JohnnyCrazy
